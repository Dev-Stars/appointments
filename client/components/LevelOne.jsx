import React from 'react';
import PropTypes from 'prop-types';

const titleStyle = {
  fontSize: '20px',
  fontWeight: '600',
};

const largeSelectable = {
  marginTop: '10px',
  width: '100%',
  fontSize: '14px',
  border: '1px solid #ccc',
  height: '34px',
  fontFamily: 'Helvetica',
};

const leftSmallSlectable = {
  marginTop: '10px',
  marginRight: '5%',
  width: '47.5%',
  fontSize: '14px',
  border: '1px solid #ccc',
  height: '34px',
  marginBottom: '10px',
};

const rightSmallSelectable = {
  marginTop: '10px',
  width: '47.5%',
  fontSize: '14px',
  border: '1px solid #ccc',
  height: '34px',
  marginBottom: '10px',
};

const findTableButton = {
  width: '100%',
  color: 'white',
  background: '#3cb52e',
  borderColor: '#28781e',
  boxShadow: '0 1px 1px rgba(0,0,0,0.3)',
  borderRadius: '5px',
  padding: '5px 8px',
  fontSize: '12px',
  fontWeight: '600',
  lineHeight: '1.5em',
  height: '34px',
};

const getGuestCount = (maximum) => {
  if (typeof maximum !== 'number' || maximum < 0) {
    return null;
  }
  const totalCount = [];
  for (let j = 1; j <= 12; j += 1) {
    if (j <= maximum) {
      totalCount.push(j);
    } else {
      break;
    }
  }
  return totalCount;
};

const LevelOne = (props) => {
  const {
    nextTwoWeeks, times, max, next, change, count, date, time,
  } = props;

  const guestCount = getGuestCount(max);

  return (
    <div id="viewOne">
      <div style={titleStyle}>
        Make a Reservation
      </div>
      <div>
        <select id="selectedData" onChange={change} name="selectedDate" value={date} style={largeSelectable}>
          {nextTwoWeeks.map(possibleDate => (
            <option value={possibleDate} key={possibleDate.toString()}>
              {possibleDate}
            </option>
          ))}
        </select>
      </div>
      <div>
        <select id="selectedTime" onChange={change} name="selectedTime" value={time} style={leftSmallSlectable}>
          {times.map(possibleTime => (
            <option value={possibleTime} key={possibleTime.toString()}>
              {possibleTime}
            </option>
          ))}
        </select>
        <select id="guestCount" onChange={change} name="guestCount" value={count} style={rightSmallSelectable}>
          {guestCount.map(possibleGuestCount => (
            <option value={possibleGuestCount} key={possibleGuestCount.toString()}>
              {possibleGuestCount}
            </option>
          ))}
        </select>
      </div>
      <button type="button" onClick={next} style={findTableButton}>
        Find a Table
      </button>
    </div>
  );
};

LevelOne.propTypes = {
  nextTwoWeeks: PropTypes.arrayOf(PropTypes.string).isRequired,
  times: PropTypes.arrayOf(PropTypes.string).isRequired,
  max: PropTypes.number.isRequired,
  next: PropTypes.func.isRequired,
  change: PropTypes.func.isRequired,
  count: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
};

export default LevelOne;
