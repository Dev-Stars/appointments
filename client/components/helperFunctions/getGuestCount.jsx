const getGuestCount = (maximum) => {
  if (typeof maximum !== 'number' || maximum < 0) {
    return null;
  }
  const totalCount = [];
  for (let j = 1; j <= 12; j += 1) {
    if (j <= maximum) {
      totalCount.push(j);
    } else {
      break;
    }
  }
  return totalCount;
};

export default getGuestCount;
