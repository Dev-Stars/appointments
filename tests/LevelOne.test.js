import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import LevelOne from '../client/components/LevelOne.jsx';
import getGuestCount from '../client/components/helperFunctions/getGuestCount.jsx';

enzyme.configure({ adapter: new Adapter() });

  describe('LevelOne, a child component', () => {
    it('should render 3 dropdown menus, and a submit button', () => {
      const levelOneComponent = shallow(
        <LevelOne
          nextTwoWeeks={["July 26, 2018", "July 27, 2018", "July 28, 2018", "July 29, 2018", "July 30, 2018", "July 31, 2018", "August 1, 2018", "August 2, 2018", "August 3, 2018", "August 4, 2018", "August 5, 2018", "August 6, 2018", "August 7, 2018", "August 8, 2018"]}
          times={["9:30 AM", "10:30 AM", "11:30 AM", "12:30 PM", "1:30 PM", "2:30 PM", "3:30 PM"]}
          max={5}
          change={(() => { null })()}
          next={(() => { null })()}
          count={'2'}
          date={'July 26, 2018'}
          time={'9:30 AM'}
        />
      );

      expect(toJson(levelOneComponent)).toMatchSnapshot();
    });

    it('getGuestCount should return an array of numbers, or null', () => {
      expect(getGuestCount(100).length).toEqual(12);
      expect(getGuestCount(50).length).toEqual(12);
      expect(getGuestCount(2).length).toEqual(2);
      expect(getGuestCount(3).length).toEqual(3);
      expect(getGuestCount(4).length).toEqual(4);
      expect(getGuestCount(-5)).toEqual(null);
      expect(getGuestCount(0).length).toEqual(0);
      expect(getGuestCount('5')).toEqual(null);
      expect(getGuestCount('')).toEqual(null);
      expect(getGuestCount()).toEqual(null);
      expect(getGuestCount(true)).toEqual(null);
      expect(getGuestCount(false)).toEqual(null);

      const arrayOfNumber = getGuestCount(12);
      const allNumbers = [];
      arrayOfNumber.forEach(countValue => {
        if (typeof countValue === 'number') {
          allNumbers.push(countValue);
        }
      });
      expect(allNumbers.length).toEqual(12);
    });

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <LevelOne
        nextTwoWeeks={["July 26, 2018", "July 27, 2018", "July 28, 2018", "July 29, 2018", "July 30, 2018", "July 31, 2018", "August 1, 2018", "August 2, 2018", "August 3, 2018", "August 4, 2018", "August 5, 2018", "August 6, 2018", "August 7, 2018", "August 8, 2018"]}
        times={["9:30 AM", "10:30 AM", "11:30 AM", "12:30 PM", "1:30 PM", "2:30 PM", "3:30 PM"]}
        max={5}
        change={(() => { console.log('dummy') })()}
        next={(() => { console.log('dummy') })()}
        count={'2'}
        date={'July 26, 2018'}
        time={'9:30 AM'}
      />
    );
  });

  it('should render one correct css wrapper', () => {
    expect(wrapper.find('div#viewOne').exists()).toEqual(true);
  });

  it('should render 3 select options and a button', () => {
    expect(wrapper.find('select#selectedDate').exists()).toEqual(true);
    expect(wrapper.find('select#selectedTime').exists()).toEqual(true);
    expect(wrapper.find('select#guestCount').exists()).toEqual(true);
    expect(wrapper.find('button').exists()).toEqual(true);
  });
});
