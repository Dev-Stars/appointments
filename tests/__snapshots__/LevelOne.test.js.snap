// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelOne, a child component should render 3 dropdown menus, and a submit button 1`] = `
<div
  id="viewOne"
>
  <div
    style={
      Object {
        "fontSize": "20px",
        "fontWeight": "600",
      }
    }
  >
    Make a Reservation
  </div>
  <div>
    <select
      id="selectedDate"
      name="selectedDate"
      style={
        Object {
          "border": "1px solid #ccc",
          "fontFamily": "Helvetica",
          "fontSize": "14px",
          "height": "34px",
          "marginTop": "10px",
          "width": "100%",
        }
      }
      value="July 26, 2018"
    >
      <option
        key="July 26, 2018"
        value="July 26, 2018"
      >
        July 26, 2018
      </option>
      <option
        key="July 27, 2018"
        value="July 27, 2018"
      >
        July 27, 2018
      </option>
      <option
        key="July 28, 2018"
        value="July 28, 2018"
      >
        July 28, 2018
      </option>
      <option
        key="July 29, 2018"
        value="July 29, 2018"
      >
        July 29, 2018
      </option>
      <option
        key="July 30, 2018"
        value="July 30, 2018"
      >
        July 30, 2018
      </option>
      <option
        key="July 31, 2018"
        value="July 31, 2018"
      >
        July 31, 2018
      </option>
      <option
        key="August 1, 2018"
        value="August 1, 2018"
      >
        August 1, 2018
      </option>
      <option
        key="August 2, 2018"
        value="August 2, 2018"
      >
        August 2, 2018
      </option>
      <option
        key="August 3, 2018"
        value="August 3, 2018"
      >
        August 3, 2018
      </option>
      <option
        key="August 4, 2018"
        value="August 4, 2018"
      >
        August 4, 2018
      </option>
      <option
        key="August 5, 2018"
        value="August 5, 2018"
      >
        August 5, 2018
      </option>
      <option
        key="August 6, 2018"
        value="August 6, 2018"
      >
        August 6, 2018
      </option>
      <option
        key="August 7, 2018"
        value="August 7, 2018"
      >
        August 7, 2018
      </option>
      <option
        key="August 8, 2018"
        value="August 8, 2018"
      >
        August 8, 2018
      </option>
    </select>
  </div>
  <div>
    <select
      id="selectedTime"
      name="selectedTime"
      style={
        Object {
          "border": "1px solid #ccc",
          "fontSize": "14px",
          "height": "34px",
          "marginBottom": "10px",
          "marginRight": "5%",
          "marginTop": "10px",
          "width": "47.5%",
        }
      }
      value="9:30 AM"
    >
      <option
        key="9:30 AM"
        value="9:30 AM"
      >
        9:30 AM
      </option>
      <option
        key="10:30 AM"
        value="10:30 AM"
      >
        10:30 AM
      </option>
      <option
        key="11:30 AM"
        value="11:30 AM"
      >
        11:30 AM
      </option>
      <option
        key="12:30 PM"
        value="12:30 PM"
      >
        12:30 PM
      </option>
      <option
        key="1:30 PM"
        value="1:30 PM"
      >
        1:30 PM
      </option>
      <option
        key="2:30 PM"
        value="2:30 PM"
      >
        2:30 PM
      </option>
      <option
        key="3:30 PM"
        value="3:30 PM"
      >
        3:30 PM
      </option>
    </select>
    <select
      id="guestCount"
      name="guestCount"
      style={
        Object {
          "border": "1px solid #ccc",
          "fontSize": "14px",
          "height": "34px",
          "marginBottom": "10px",
          "marginTop": "10px",
          "width": "47.5%",
        }
      }
      value="2"
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
    </select>
  </div>
  <button
    style={
      Object {
        "background": "#3cb52e",
        "borderColor": "#28781e",
        "borderRadius": "5px",
        "boxShadow": "0 1px 1px rgba(0,0,0,0.3)",
        "color": "white",
        "fontSize": "12px",
        "fontWeight": "600",
        "height": "34px",
        "lineHeight": "1.5em",
        "padding": "5px 8px",
        "width": "100%",
      }
    }
    type="button"
  >
    Find a Table
  </button>
</div>
`;
